Lib9c/Action/BattleArena.cs

```diff
@@ -223,7 +224,27 @@ public override IAccountStateDelta Execute(IActionContext context)
                 arenaInformation.ResetTicket(currentTicketResetCount);
             }
 
-            arenaInformation.UseTicket(ticket);
+            if (arenaInformation.Ticket > 0)
+            {
+                if (roundData.ArenaType != ArenaType.OffSeason && ticket > 1)
+                {
+                    throw new ExceedPlayCountException($"[{nameof(BattleArena)}] ticket : {ticket}");
+                }
+
+                arenaInformation.UseTicket(ticket);
+            }
+            else
+            {
+                if (ticket > 1)
+                {
+                    throw new ExceedTicketPurchaseCountException($"[{nameof(BattleArena)}] ticket : {ticket}");
+                }
+
+                var ticketBalance = ArenaHelper.GetTicketPrice(roundData, arenaInformation, states.GetGoldCurrency());
+                var arenaAdr = ArenaHelper.DeriveArenaAddress(roundData.ChampionshipId, roundData.Round);
+                states = states.TransferAsset(context.Signer, arenaAdr, ticketBalance);
+                arenaInformation.BuyTicket();
```

Comment:
is this correct? i can't found use ticket in this case.

Reply:
We can only buy one ticket. So I didn't create any parameters in Buy Ticket()